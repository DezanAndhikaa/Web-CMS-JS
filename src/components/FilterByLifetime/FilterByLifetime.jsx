import React from 'react';
import './FilterByLifetime.scss';
import CloseButton from '../../components/CloseButton/CloseButton';
import { Button } from '@material-ui/core';

class FilterByLifetime extends React.Component{

    constructor(props){
        super(props);
        this.state ={
            value1: '',
            value2: ''
        }
    }

    handleChange = (event) => {
        event.preventDefault();
        const { name, value } = event.target;
        this.setState({
            [name] : value
        })
    }

    render(){
        return(
            <div className="assign-lifetime-modal">
                <div className="top-row">
                    <div className="ut-underline"/>
                    <p className="select-input-title">{this.props.title}</p>
                    <CloseButton onClose={this.props.onClosed}/>
                </div>
                <div className="teks-middle">
                    <label className="teks-left">Min</label>    
                    <label className="teks-right">Max</label>
                </div>
                <div className="top-midel">
                    <input className="input-left" name="value1" onChange={this.handleChange} value={this.state.value1}></input>
                    <input className="input-right" name="value2" onChange={this.handleChange} value={this.state.value2}></input>
                </div>
                
                <div className="bottom-rows">
                    <Button className="btn-search" onClick={ () => {this.props.onFilter(this.state.value1, this.state.value2); this.props.onClosed()}} >Search Lifetime</Button>
                </div>
            </div>
        )
    }
}

export default FilterByLifetime