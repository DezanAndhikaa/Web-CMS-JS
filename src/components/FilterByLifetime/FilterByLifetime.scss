@import '../../themes/functions';
@import '../../themes/variables';

.assign-lifetime-modal{
    position: 'absolute';
    width: vw(400);
    height: vw(155);
    background-color: white;
    margin: auto;
    border-radius: vw(20);
    display: flex;
    justify-content: space-between;
    flex-direction: column;
    padding: vw(30);
    .ut-underline{
        margin-top: vw(1.5);
        margin-bottom: vw(10);
        width: vw(8);
        height: vw(20);
        background-color: map-get($colors, primary);
    }
    .top-row{
        display: flex;
        justify-content: space-between;
        height: vw(30);
    }
    .teks-middle{
        margin-top: vw(10);
        display: flex;
        .teks-left{
            display: flex;
            justify-content: center;
            font-size: 12px;
        }
        .teks-right{
            margin-left: 48.5%;
            font-size: 12px;
            display: flex;
            justify-content: center;
        }
    }
    .top-midel{
        margin-top: vw(5);
        display: flex;
        height: vw(100);
        .input-left{
            display: flex;
            justify-content: center;
            width: vw(130);
            height: vw(30);

            background: #fafafa;
            -webkit-box-shadow: inset 0px 1px 3px 0px rgba(0, 0, 0, 0.08);
            -moz-box-shadow: inset 0px 1px 3px 0px rgba(0, 0, 0, 0.08);
            box-shadow: inset 0px 1px 3px 0px rgba(0, 0, 0, 0.08);
            -webkit-border-radius: 5px;
            -moz-border-radius: 5px;
            border-radius: 5px;
            padding: 0 10px;
            font-size: 16px;
            color: #666;
            -webkit-transition: all 0.4s ease;
            -o-transition: all 0.4s ease;
            -moz-transition: all 0.4s ease;
            transition: all 0.4s ease;
            line-height: 50px;
        }
        .input-right{
            margin-left: 15%;
            width: vw(130);
            height: vw(30);

            background: #fafafa;
            -webkit-box-shadow: inset 0px 1px 3px 0px rgba(0, 0, 0, 0.08);
            -moz-box-shadow: inset 0px 1px 3px 0px rgba(0, 0, 0, 0.08);
            box-shadow: inset 0px 1px 3px 0px rgba(0, 0, 0, 0.08);
            -webkit-border-radius: 5px;
            -moz-border-radius: 5px;
            border-radius: 5px;
            padding: 0 10px;
            font-size: 16px;
            color: #666;
            -webkit-transition: all 0.4s ease;
            -o-transition: all 0.4s ease;
            -moz-transition: all 0.4s ease;
            transition: all 0.4s ease;
            line-height: 50px;
        }
    }
    
    .select-input-title{
        margin: 0;
        margin-right: vw(17);
        font-weight: bold;
        font-size: 15px; 
        width: 60%;
    }

    .bottom-rows{
        display: flex;
        margin-top: vw(20);
        .btn-search{
            display: flex;
            align-items: center;
            justify-content: center;
            margin-left: 60%;
            width: 35%;
            height: vw(33.07) !important;
            font-size: vw(13) !important;
            font-weight: bold;
            line-height: vw(16) !important;
            font-family: 'Lato-Regular';
            background-color: map-get($colors , primary);
            text-transform: none;
            box-shadow: 0px vw(3) vw(6) rgba(0, 0, 0, 0.161);
            border-radius: 0.4vw;
            cursor: pointer;
        }
        .btn-search:hover {
            background-color: map-get($colors, primary-dark);
        }
    }
}

