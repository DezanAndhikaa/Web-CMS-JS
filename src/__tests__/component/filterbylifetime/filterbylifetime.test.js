import React from 'react';
import Enzyme, { shallow } from "enzyme";
import Adapter from 'enzyme-adapter-react-16';
import FilterByLifetime from '../../../components/FilterByLifetime/FilterByLifetime';

Enzyme.configure({adapter:new Adapter()});
it("render filter by lifetime correctly", () => {
    const wrapper = shallow (
        <FilterByLifetime />
    )
    expect(wrapper).toMatchSnapshot();
})