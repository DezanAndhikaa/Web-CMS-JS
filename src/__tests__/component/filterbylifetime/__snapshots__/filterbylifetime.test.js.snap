// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render filter by lifetime correctly 1`] = `ShallowWrapper {}`;
